calblueprint/unloop
# Implement Assignment controller/API controller

Implemented the Assignment Controller & API Controller
## Assignment API Controller ##
- I didn't know if it was best to create two controllers, one for CRUD operations on ActionItems and one for CRUD operations on Assignments (that could/would call the ActionItem API controller as well). Right now they're in one controller, but I wouldn't mind separating the logic if that's what's best. I personally think there's too much logic in the controller right now.

- I don't like the idea of using `Assignments` so much when the colloquial term for this feature is an `Action Item`. I would ideally want to rename our current `Assignments` model to `ActionItems` and rename our current `ActionItems` model to `ActionItemsInfo`, since then we would be making calls to an `ActionItemController` when creating an action item instead of an `AssignmentsController`. If this is a better layout (or if you have feedback on this proposed layout) I'd like to hear about it and I can make the refactor some time during Spring Break.

## Assignment Controller ##
- Currently the `edit`, `index`, and `show` views work (though they're ugly) and the `new` view is purposefully left empty. I checked and no models have a `new` view. I did create logic in the controller for `new` but it doesn't do anything ATM. I did this just because the `paperworks` and `case_notes` controllers both have logic for a `new` view but no `new` view.

## To Test ##
- Navigate through the `assignments/:id/edit`, `assignments/`, `assignments/:id/show` views to make sure they work properly. 
- To test the API controller, I created a testing component called `AssignmentButton` and placed it in the `/assignments` view. The `title/description` values the button creates/updates and the assignment `ids` it does updates/deletes on are hardcoded, so you're going to have to go to `components/AssignmentButton` to change the values if you wish to do so. The button console logs the response to the request it makes so you're free.
- Before the PR gets merged we should definitely remember to delete the AssignmentButton!


CC: @didvi
